File: app/src/main/java/com/example/devfest/AuthorsAdapter.java
package com.example.devfest;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

public class AuthorsAdapter
        extends RecyclerView.Adapter<AuthorsAdapter.AuthorsHolder> {

    private final List<Speaker> authors;

    AuthorsAdapter(List<Speaker> authors) {
        this.authors = authors;
    }

    @NonNull
    @Override
    public AuthorsHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View itemView = inflater.inflate(R.layout.item_speaker, parent, false);
        return new AuthorsHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull AuthorsHolder holder, int position) {
        Speaker speaker = authors.get(position);
        holder.photoView.setImageResource(speaker.getPhoto());
        holder.nameView.setText(speaker.getName());
        holder.profView.setText(speaker.getProf());
        holder.cityView.setText(speaker.getCity());
    }

    @Override
    public int getItemCount() {
        return authors.size();
    }

    static class AuthorsHolder extends RecyclerView.ViewHolder {
        final ImageView photoView;
        final TextView nameView, profView, cityView;
        AuthorsHolder(View itemView){
            super(itemView);
            photoView = (ImageView)itemView.findViewById(R.id.photo);
            nameView = (TextView)itemView.findViewById(R.id.name);
            profView = (TextView)itemView.findViewById(R.id.prof);
            cityView = (TextView)itemView.findViewById(R.id.city);
        }
    }
}

File: app/src/main/java/com/example/devfest/authorsAdapter.kt
package com.example.devfest

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView

class AuthorsAdapter2 internal constructor(private val authors: List<Author>) :
    RecyclerView.Adapter<AuthorsAdapter.AuthorsHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AuthorsHolder {
        // create a new view
        val inflater = LayoutInflater.from(parent.context)
        val itemView = inflater.inflate(R.layout.item_speaker, parent, false)
        return AuthorsHolder(itemView)
    }

    override fun onBindViewHolder(holder: AuthorsHolder, position: Int) {
        // get our custom object from our dataset at this position
        val author = authors[position]
        holder.photoView.setImageResource(Catalog.getPhoto())
        holder.nameView.setText(Catalog.getName())
        holder.profView.setText(Catalog.getProf())
        holder.cityView.setText(Catalog.getCity())
    }

    override fun getItemCount(): Int {
        return authors.size
    }

    internal class AuthorsHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val photoView: ImageView
        val nameView: TextView
        val profView: TextView
        val cityView: TextView

        init {
            photoView = itemView.findViewById(R.id.photo) as ImageView
            nameView = itemView.findViewById(R.id.name) as TextView
            profView = itemView.findViewById(R.id.prof) as TextView
            cityView = itemView.findViewById(R.id.city) as TextView
        }
    }
}

